MenuItem.cs

```diff
@@ -336,6 +381,7 @@ protected override void OnCheckedChanged(EventArgs e)
 
         private IGH_DocumentObject[] _visibleObjects;
         private IGH_DocumentObject[] _filteredObjects;
+
         private SortedDictionary<string, string> _nicknamesCache = new SortedDictionary<string, string>();
```

Comment:
In your case I believe `Dictionary` should be faster than `SortedDictionary` as `SortedDictionary` is a BST while `Dictionary` is a hashtable (the sorted version has an overhead of maintaining the order of keys). But I'll keep it here since the bottleneck would be `String.Join`.